src/posix/client.cpp

```diff
@@ -128,14 +131,46 @@ int main(int argc, char *argv[])
         strncpy(sockname.sun_path, OPENTHREAD_POSIX_DAEMON_SOCKET_NAME, sizeof(sockname.sun_path) - 1);
 
         ret = connect(sSessionFd, reinterpret_cast<const struct sockaddr *>(&sockname), sizeof(struct sockaddr_un));
+    }
 
-        if (ret == -1)
-        {
-            fprintf(stderr, "OpenThread daemon is not running.\n");
-            ExitNow(ret = OT_EXIT_FAILURE);
-        }
+exit:
+    if (ret == -1 && sSessionFd != -1)
+    {
+        close(sSessionFd);
+        sSessionFd = -1;
+    }
+    return ret;
+}
+
+static bool ReconnectSession(void)
+{
+    bool     ok    = false;
+    uint32_t delay = 100000; // 100ms
+
+    for (int i = 0; i < 5; i++) // delay for 3.1s in total
+    {
+        usleep(delay);
+
+        VerifyOrExit(ConnectSession() == -1, ok = true);
```

Comment:
print error information of each failure?

Reply:
The current behavior is mostly like OT CLI. 

Note that the first delay is 100ms. It is possible that ot-daemon hasn't finished restarting in 100ms, so I would prefer `ot-ctl` to simply wait for more time without printing any error. 

Furthermore, if all 5 reconnecting retry failed, it's very likely that ot-daemon has really exited, instead of being reset. In such case, I am fine to print error before `ot-ctl` exits, but exiting quietly should also be fine?